pkg/reconciler/clusterimagepolicy/controller.go

```diff
@@ -17,25 +17,57 @@ package clusterimagepolicy
 import (
 	"context"
 
+	"k8s.io/client-go/tools/cache"
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	configmapinformer "knative.dev/pkg/client/injection/kube/informers/core/v1/configmap"
+	secretinformer "knative.dev/pkg/client/injection/kube/informers/core/v1/secret"
 	"knative.dev/pkg/configmap"
 	"knative.dev/pkg/controller"
+	"knative.dev/pkg/logging"
+	pkgreconciler "knative.dev/pkg/reconciler"
 
+	"github.com/sigstore/cosign/pkg/apis/config"
 	clusterimagepolicyinformer "github.com/sigstore/cosign/pkg/client/injection/informers/cosigned/v1alpha1/clusterimagepolicy"
 	clusterimagepolicyreconciler "github.com/sigstore/cosign/pkg/client/injection/reconciler/cosigned/v1alpha1/clusterimagepolicy"
 )
 
+const SystemNamespace = "cosign-system"
+
 // NewController creates a Reconciler and returns the result of NewImpl.
 func NewController(
 	ctx context.Context,
 	cmw configmap.Watcher,
 ) *controller.Impl {
 	clusterimagepolicyInformer := clusterimagepolicyinformer.Get(ctx)
+	cmInformer := configmapinformer.Get(ctx)
+	secretInformer := secretinformer.Get(ctx)
 
-	r := &Reconciler{}
+	r := &Reconciler{
+		secretlister:    secretInformer.Lister(),
+		configmaplister: cmInformer.Lister(),
+		kubeclient:      kubeclient.Get(ctx),
+	}
 	impl := clusterimagepolicyreconciler.NewImpl(ctx, r)
-	r.Tracker = impl.Tracker
+	r.tracker = impl.Tracker
 
 	clusterimagepolicyInformer.Informer().AddEventHandler(controller.HandleAll(impl.Enqueue))
 
+	// When the underlying ConfigMap changes,perform a global resync on
+	// ClusterImagePolicies to make sure their state is correctly reflected
+	// in the ConfigMap. This is admittedly a bit heavy handed, but I don't
+	// really see a way around it, since if something is wrong with the
+	// ConfigMap but there are no changes to the ClusterImagePolicy, it needs
+	// to be synced.
+	grCb := func(obj interface{}) {
+		logging.FromContext(ctx).Info("Doing a global resync on ClusterImagePolicies due to ConfigMap changing.")
+		impl.GlobalResync(clusterimagepolicyInformer.Informer())
+	}
+	// Resync on only ConfigMap changes that pertain to the one I care about.
+	cmInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: pkgreconciler.ChainFilterFuncs(
+			pkgreconciler.NamespaceFilterFunc(SystemNamespace),
+			pkgreconciler.NameFilterFunc(config.ImagePoliciesConfigName)),
+		Handler: controller.HandleAll(grCb),
+	})
```

Comment:
I thought we had a way to configure this on the `configmap.Watcher`? 🤔

Reply:
There probably is :) I'll go look, this one I was fairly sure would work :)

Reply:
Since I need the Lister anyways to get the raw CM, decided to just leave it and it works just fine. LMK if you feel strongly and I can rejigger.

Reply:
Nope, easy enough to change if we need to.